.buildkite/scripts/install-kubectl.sh

```diff
@@ -34,7 +34,7 @@ else
     ARCH_SUFFIX=amd64
 fi
 
-if curl -sSLo "${KUBECTL_CMD}" "https://storage.googleapis.com/kubernetes-release/release/${K8S_VERSION}/bin/${OS}/${ARCH_SUFFIX}/kubectl" ; then
+if curl -sSLo "${KUBECTL_CMD}" "https://dl.k8s.io/release/${K8S_VERSION}/bin/${OS}/${ARCH_SUFFIX}/kubectl"; then
```

Comment:
For context, see https://github.com/elastic/elastic-package/pull/2502#issuecomment-2766874341.